Source/Core/Core/HW/CPU.cpp

```diff
@@ -63,6 +67,34 @@ void CPUManager::ExecutePendingJobs(std::unique_lock<std::mutex>& state_lock)
   }
 }
 
+void CPUManager::StartTimePlayedTimer()
+{
+  // Steady clock for greater accuracy of timing
+  std::chrono::steady_clock timer;
+  auto prev_time = timer.now();
+
+  while (true)
+  {
+    const std::string game_id = SConfig::GetInstance().GetGameID();
+    TimePlayed time_played(game_id);
+    auto curr_time = timer.now();
+
+    // Check that emulation is not paused
+    if (m_state == State::Running)
+    {
+      auto diff_time = std::chrono::duration_cast<std::chrono::milliseconds>(curr_time - prev_time);
+      time_played.AddTime(diff_time);
+    }
+
+    prev_time = curr_time;
+
+    if (m_state == State::PowerDown)
+      return;
+
+    m_time_played_finish_sync.WaitFor(std::chrono::seconds(30));
```

Comment:
This thread will run code every 30 seconds until `m_state` becomes `PowerDown`. This is a problem on Android, where power consumption is a concern, because this will force the CPU to be awake every 30 seconds even if Dolphin is paused and the device's screen is off. When one of the following two conditions are triggered (you can choose which one of the two you want to implement), please either shut down the thread or make the thread wait indefinitely on e.g. a condition variable:

1. `m_state` becomes `Stepping`
2. [setBackgroundExecutionAllowedNative](https://github.com/dolphin-emu/dolphin/blob/3ea870ef8c5f4d4b86bd3ce0c2e3ddba6f1f25aa/Source/Android/jni/ActivityTracker.cpp#L12) is called with `allowed == false`